### ISSUE TITLE
Fix `no-invalid-position-declaration` false positives for embedded styles

### ISSUE THREAD

**Issue by reduckted** [stylelint/stylelint#8695]
### What minimal example or steps are needed to reproduce the bug?

```html
<div style="color: red"></div>
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "overrides": [
    { "files": ["**/*.html"], "customSyntax": "postcss-html" }
  ],
  "rules": {
    "no-invalid-position-declaration": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8UCWA3ABAZwC4E8AbGAXgB1wJCIAneTGmKCgPiQHo10WyA7EADQgAZqmIA5AIYBbOIhgAPGQAdiAOgAWuaYUGVeogOYIQwPpkwUI6GDRqpY2CvQDa5i5mCWRYmE4SYbiAAVMHswZrahBQAugLeYACueBDSAMr4vLiSCs7eyhB4YNjYALRaOhSYAL7uce4UNInE-vRmvB7evBClqLzokoQOpQXYqLioELylsGCEkjSSE1N5uE0w7rW81XqQBqiGAGK00ksmAFbYU3qwytgm7RYUeEQwQ1l5FACMAGxqAEwAZjUAAYKAIGiAXsR3rhSnsjKU8JJeFAFswAhR5rg-LhwZDRrhimUKtFMSBsbiKHwdtUgA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.23.0",
  "postcss-html": "latest"
}
```

### What did you expect to happen?

No problem.

### What actually happened?

`color: red` is reported as a problem.

**Comment by Mouvedia** [comment]
@sw1tch3roo

**Comment by jeddy3** [comment]
@reduckted Thank you for the report and the pull request.

This also impacts other embedded styles, such as styled css-in-js:

```js
const foo = styled.a`color: red;`
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBc4IDsDOAXABAMwhTAvJhgJ4A2MUAdAIYAGkZEATvJs5QNx0gA0I2AJYUAcjQC2cRDAAeEgA4UqAK1R8kyIQHMEIYAB1kmTPvABXDBHEBlEsnQ0ZptqfkQMYVKgC0pClF87Byc+Q2NTZjMKVGdMAyNjExBkCG9BZAA3GjJBALdUQXRBFG9YMDIaZhoilFj0SJgwzABfQ2b1SE1BLQAxFnFq3VUUdVh5NUR48JA-GBz7WNMK9BgMU14m13d0Tx9ZgNQgx0WQZdX0UzaQZqA).
